### ISSUE TITLE
Wrong address count in listing when fake instructions are used.

### ISSUE THREAD

**Issue by maziac** [z00m128/sjasmplus#25]
The following small code

    ORG 0x8000
    ld hl,0x8123 
    sub a,2
    add a,a

produces the following list file:

024   0000             
025   0000                 ORG 0x8000
026   8000             
027   8000 21 23 81        ld hl,0x8123 
028   8004 97 D6 02        sub a,2
029   8006 87              add a,a
030   8007             

Please have a look at line 28. The address is 8004 but it should be 8003.
Then at line 29 the addresses are OK again.

I'm not sure what is happening here. I have entered 'sub a,2' by mistake, I meant 'sub 2' but the assembler seems to make something meaningful of it. I.e. I tried:
`sub c,2`
which results into

   sub c
   sub 2

Is this some kind of pseudo instruction?

Anyhow, even if so the line numbering in the list file is wrong.

**Comment by z00m128** [comment]
Hello, this is a documented feature: 

>     and all other commands support this.
>
>    LD A,B,B,D,D,H
>   /* this is:
>    LD A,B
>     LD B,D
>     LD D,H
>   */
>   ;or you can write  LD A,B:LD B,D:LD D,H

If you pay notice to the listing, the `sub a,2` is compiled as `sub a: sub 2`. We agree, that there should be a warning on fake instructions, as it is requested in Issue #7, therefore I'm closing it as duplicity.

**Comment by maziac** [comment]
Hi,

could you please open this again.
I'm perfectly fine with the fake instructions. 

My main point was about the addresses in the listing (list file).
These are wrong: LD HL,nn starts at 8000h and is 3 byte but the next address is 8004h and not 8003h.
Maybe the address of the last instruction of the line is used but I think it should be the address of the first instruction of a line.

**Comment by z00m128** [comment]
I'm reopening it as you requested, but there are more troubles with parser. I guess they will be fixed at once, maybe with issue #7.  I tried the example from docs and it's really more than wierd:
```
1   0000                org 32768
2   8000             
3   8002 78 42 54    start:     LD A,B,B,D,D,H
4   8003 C9             ret
5   8004             
6   8004

**Comment by z00m128** [comment]
Fixed in https://github.com/z00m128/sjasmplus/commit/b6911ce9b8b04d4b79d8ae395fe90556197a08f9 by Busy.
